lib/ex_pwned/api.ex

```diff
@@ -18,10 +18,23 @@ defmodule ExPwned.Api do
         do: do_get(path_arg, Enum.into(query_params, %{}))
 
       def do_get(path_arg, query_params) do
-        path_arg
-        |> build_url(query_params)
-        |> HTTPoison.get(headers())
-        |> Parser.parse()
+        {:ok, result, status} =
```

Comment:
seems like this matching will not work for error case that return error tuple.

Reply:
Question for you.  I have asked a few people online and their recommendation is to not retry in the library, but to error and emit the retry timeout.  Letting the application using the library to determine how to handle the response.  Does this sound like a better approach to you?

Reply:
That is definitely something I agree with. Just need to emit the timeout error information that includes timeout value before another retry.

Reply:
Let's do this.  I will send a different PR for the new approach